packages/parallel/init.lua

```diff
@@ -3,65 +3,394 @@ local base = require("packages.base")
    end
-   SILE.typesetter = oldtypesetter
+   SILE.typesetter = oldtypesetter -- Restore the original typesetter after iteration.
 end
 
+-- A null typesetter used as a placeholder. This typesetter doesn't output any content.
+-- Its purpose is to make the transtion between frames easier and trouble free.
 local nulTypesetter = pl.class(SILE.typesetters.base) -- we ignore this
-nulTypesetter.outputLinesToPage = function () end
-
-local parallelPagebreak = function ()
-   for i = 1, #folioOrder do
-      local thisPageFrames = folioOrder[i]
-      for j = 1, #thisPageFrames do
-         local frame = thisPageFrames[j]
-         local typesetter = typesetterPool[frame]
+nulTypesetter.outputLinesToPage = function() end
+
+-- Utility function to calculate the height of new material for a given frame.
+-- This function computes the cumulative height by adding the heights of lines
+-- in the `typesetter.state.outputQueue` starting from the marked position.
+local calculateFrameHeight = function(frame, typesetter)
+   -- Retrieve the cumulative height for the frame or initialize it if not set.
+   local height = calculations[frame].cumulativeHeight or SILE.types.length()
+
+   -- Iterate through the output queue from the marked position, summing line heights.
+   -- Each line's height is the sum of its height and depth.
+   for i = calculations[frame].mark + 1, #typesetter.state.outputQueue do
+      local lineHeight = typesetter.state.outputQueue[i].height + typesetter.state.outputQueue[i].depth
+      height = height + lineHeight
+   end
+
+   -- Return the updated cumulative height for the frame.
+   return height
+end
+
+-- Calculate the line height of a sample text.
+-- This function uses a sample text containing two characters:
+-- one with an ascender and one with a descender.
+-- The height calculation includes the glyph heights and the baseline skip setting.
+local calculateLineHeight = function(sampleText)
+   local glyphs = SILE.shaper:shapeToken(sampleText, SILE.font.loadDefaults({}))
+   local lineSkip = SILE.settings:get("document.lineskip").height
+   return glyphs[1].height + glyphs[2].depth + lineSkip:tonumber()
+end
+
+-- Create dummy content to fill up the overflowed frames.
+local createDummyContent = function(height, frame, offset)
+   -- Get the typesetter for the frame
+   local typesetter = typesetterPool[frame]
+
+   -- Calculate precise line height by typesetting a sample line
+   -- local lineHeight = calculateLineHeight("hp")
+
+   -- log("Precise lineHeight after simulation = ", lineHeight)
+
+   -- If lineHeight could not be calculated, fall back to baselineSkip and lineSkip of the document
+   if not lineHeight then
+      local baselineSkip = SILE.settings:get("document.baselineskip").height or SILE.types.length({ length = 0 })
+      local lineSkip = SILE.settings:get("document.lineskip").height or SILE.types.length({ length = 0 })
```

Comment:
The problem here is that depending on the linespacing algorithm (the default one is more or less the TeX one), the lineskip is used conditionally (when boxes are greater than the baselineskip and would collide vertically), not always. In most cases (i.e. regular text with decent settings for the baselineskip), the lineskip isn't applied.